lib/train-winrm/transport.rb

```diff
@@ -102,24 +107,40 @@ def connection(state = nil, &block)
       def validate_options(opts)
         super(opts)
 
-        # set scheme and port based on ssl activation
-        scheme = opts[:ssl] ? "https" : "http"
-        port = opts[:port]
-        port = (opts[:ssl] ? 5986 : 5985) if port.nil?
-        winrm_transport = opts[:winrm_transport].to_sym
+        # Normalize to symbols
+        opts[:winrm_transport] = opts[:winrm_transport].to_s.downcase.to_sym if opts[:winrm_transport]
+        opts[:winrm_shell_type] = opts[:winrm_shell_type].to_s.downcase.to_sym if opts[:winrm_shell_type]
+
+        winrm_transport = opts[:winrm_transport]
+        winrm_shell_type = opts[:winrm_shell_type]
+
         unless SUPPORTED_WINRM_TRANSPORTS.include?(winrm_transport)
           raise Train::ClientError, "Unsupported transport type: #{winrm_transport.inspect}"
         end
 
-        winrm_shell_type = opts[:winrm_shell_type].to_sym
         unless SUPPORTED_WINRM_SHELL_TYPES.include?(winrm_shell_type)
           raise Train::ClientError, "Unsupported winrm shell type: #{winrm_shell_type.inspect}"
         end
 
-        # remove leading '/'
+        # Set scheme, port, endpoint
+        scheme = opts[:ssl] ? "https" : "http"
+        port = opts[:port] || (opts[:ssl] ? 5986 : 5985)
```

Comment:
@sathish-progress Do we need to check whether transport is WinRM while assigning the WinRM SSL port?

Reply:
This is train-winrm target right it is expected to connect via winrm.. or do you have any scenario which can use train-winrm to connect to different protocols?

Reply:
Make sense